cli/translate/yaml/yaml.go

```diff
@@ -161,3 +191,123 @@ func (y *yamlTranslator) translateLoad(m yaml.MapSlice) string {
 
 	return strings.Join(values, newLineSeparator)
 }
+
+func (y *yamlTranslator) translateDeps(m []interface{}) string {
+	output := ""
+	for _, dep := range m {
+		depString, ok := dep.(string)
+		if !ok {
+			log.Printf("unknown dep type: %T", dep)
+			continue
+		}
+		module, resp, err := add.FetchModuleOrDisambiguate(strings.Split(depString, "@")[0])
+		if err != nil {
+			log.Printf("error fetching module: %s", err)
+			continue
+		}
+		output += add.GenerateSnippet(module, resp)
+
+	}
+	return output
+}
+
+func (y *yamlTranslator) translateBazel(m yaml.MapSlice) {
+	for _, field := range m {
+		key, ok := field.Key.(string)
+		if !ok {
+			log.Printf("unknown bazel field: %T", field.Key)
+			continue
+		}
+		value, ok := field.Value.(string)
+		if !ok {
+			log.Printf("unknown bazel field: %T", field.Value)
+			continue
+		}
+		if key == "version" {
+			err := os.WriteFile(".bazelversion", []byte(value), 0644)
+			if err != nil {
+				log.Printf("error writing .bazelversion file: %s", err)
+			}
+		}
+	}
+}
+
+func (y *yamlTranslator) translateSettings(m yaml.MapSlice) {
+	bazelrc := ""
+	for _, s := range m {
+		bazelrc += "common --" + s.Key.(string) + "=" + s.Value.(string) + "\n"
+	}
+
+	err := os.WriteFile(".bazelrc", []byte(bazelrc), 0644)
+	if err != nil {
+		log.Printf("error writing .bazelrc file: %s", err)
+	}
+}
+
+func (y *yamlTranslator) translateTemplateMap(m yaml.MapSlice) {
+	for _, t := range m {
+		renderTemplate(t.Value.(string), t.Key.(string))
+	}
+}
+
+func (y *yamlTranslator) translateTemplate(m []interface{}) {
+	for _, s := range m {
+		from := ""
+		into := ""
+
+		if slice, ok := s.(string); ok {
+			from = slice
+		}
+
+		if slice, ok := s.(yaml.MapSlice); ok {
+			for _, x := range slice {
+				key, ok := x.Key.(string)
+				if !ok {
+					log.Printf("unknown key type: %T", x.Value)
+					continue
+				}
+				value, ok := x.Value.(string)
+				if !ok {
+					log.Printf("unknown value type: %T", x.Value)
+					continue
+				}
+				if key == "from" {
+					from = value
+				}
+				if key == "into" {
+					into = value
+				}
+			}
+		}
+
+		if into == "" {
+			parts := strings.Split(from, "/")
+			into = parts[len(parts)-1]
+		}
+		renderTemplate(from, into)
+	}
+}
+
+func renderTemplate(from, into string) {
+	if _, err := os.Stat(into); !os.IsNotExist(err) {
+		log.Printf("skipping template %s, directory %q already exists", from, into)
+		return
+	}
+	log.Printf("grabbing template from %s and putting it into %q", from, into)
+
+	if !strings.HasPrefix(from, "github/") {
+		log.Printf("unknown template %s", from)
+	}
+
+	repo := strings.Replace(from, "github/", "https://github.com/", 1)
```

Comment:
The `github/` notation is convenient when adding deps, but I feel like requiring it to be written out as `github.com` might make the YAML format overall more convenient to work with from a usability perspective. (The URLs can be used directly instead of needing a special tool to parse them or having to manually add ".com" if you want to inspect a dependency in the browser)

Reply:
I'd like to support both formats in both places, I've added a note for myself to do this in a follow-up